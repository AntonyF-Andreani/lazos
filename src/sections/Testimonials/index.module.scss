.root {
  height: 100vh;
  width: 100%;
  position: relative;
  text-align: center;
  margin: 20px 0 0;
  display: inline-flex;
  flex-flow: row;
  flex-wrap: nowrap;
  justify-content: center;

  @media (max-width: 375px) {
    height: 140vh;
  }

  @media (max-width: 640px) and (orientation: landscape) {
    height: 100%;
  }
}

.container {
  background-color: rgb(247 247 247 / 0.5);
  display: flex;
  height: 100%;
  position: relative;
  top: 10%;
  flex-flow: row;
  align-items: center;
  justify-content: flex-end;
  flex-wrap: nowrap;
  align-content: flex-end;

  @media (max-width: 425px) {
    justify-content: center;
  }

  @media (max-width: 640px) and (orientation: landscape) {
    padding: 0;
  }
}

.mainBanner {
  position: absolute;
  height: 100%;
}

.slide {
  padding-right: 15px;
  padding-left: 15px;
}

.swiperContainer {
  height: 100%;
  padding-bottom: 40px;
}

.comentarioContainer {
  width: 100%;

  @media (min-width: 992px) {
    width: 50%;
  }

  @media (min-width: 1200px) {
    max-width: 1140px;
  }

  @media (min-width: 992px) {
    max-width: 960px;
  }

  @media (min-width: 768px) {
    max-width: 720px;
  }

  @media (min-width: 576px) {
    max-width: 540px;
  }
}

.title {
  font-weight: 700;
  font-size: 1.8em;
  line-height: 42px;
  color: #212529;
  letter-spacing: -0.05em;
  display: block;
  text-align: left;
  margin-block: 15px;
  margin-left: 9.6px;
  margin-left: 0.6rem;
}

.comentario {
  background: #b6040b;
  border-radius: 5px;
  border: 0;
  margin: 0;
  padding: 2.5rem;
  position: relative;
  font-family: "Work Sans", sans-serif;
  text-align: left;

  &::before {
    left: 22px;
    top: 21px;
    color: #fff;
    content: "â€œ";
    font-size: 85px;
    font-style: normal;
    line-height: 1;
    position: absolute;
  }
}

.comentarioText {
  color: #fff;
  font-family: "Work Sans", sans-serif;
  font-size: 1.2em;
  line-height: 26px;
  margin: 0 0 20px;
}

.autor {
  color: #111;
  display: block;
  margin-bottom: 2px;
  font-size: 16px;
  font-size: 1rem;
  text-align: left;
  line-height: 1.2;
  margin-left: 9.6px;
  margin-left: 0.6rem;
  font-weight: bolder;

  &::before {
    content: "";
    display: block;
    border-left: 11px solid transparent;
    border-right: 11px solid transparent;
    border-top: 8px solid #b6040b;
    height: 0;
    margin: 0 0 0 45px;
    width: 0;
  }
}

.lugar {
  color: #666;
  display: block;
  font-size: 12.8px;
  font-size: 0.8rem;
  font-weight: 600;
  text-align: left;
  line-height: 1.2;
  margin-left: 9.6px;
  margin-left: 0.6rem;
}

.bullet {
  display: inline-block;
  width: 10px;
  height: 10px;
  margin: 5px 7px;
  border-radius: 30px;
  background: #d6d6d6;
  cursor: pointer;

  &:hover {
    background-color: #0074ad;
  }
}

.bulletActive {
  background-color: #0074ad;
}
