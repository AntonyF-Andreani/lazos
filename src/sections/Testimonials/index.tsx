import React from "react";
import { Swiper as SwiperComponent, SwiperSlide } from "swiper/react";
import SwiperCore, { Autoplay, Pagination } from "swiper";
import { Box, Grid, Typography } from "@mui/material";
import { LazyLoadImage } from "react-lazy-load-image-component";

import styles from "./index.module.scss";

import type { Image, ClientTestimonials } from "services/Cms/types";
import Markdown from "components/MarkDown";

export interface TestimonialsProps {
  title: string;
  image: Image;
  data: ClientTestimonials[];
}

SwiperCore.use([Autoplay, Pagination]);

const Testimonials = ({ title, image, data }: TestimonialsProps) => {
  return (
    <Grid className={styles.root} component="section">
      <Box className={styles.mainBanner}>
        <LazyLoadImage
          alt={image.alt ? image.alt : "Testimonials image"}
          className={styles.image}
          effect="blur"
          src={image.filename}
        />
      </Box>
      <Grid container className={styles.container}>
        <Box className={styles.comentarioContainer}>
          <Typography className={styles.title} variant="h2">
            {title}
          </Typography>
          <Grid container>
            <SwiperComponent
              loop
              autoplay={{
                delay: 3000,
                disableOnInteraction: false,
              }}
              centeredSlides={true}
              className={styles.swiperContainer}
              pagination={{
                clickable: true,
                type: "bullets",
                bulletClass: styles.bullet,
                bulletActiveClass: styles.bulletActive,
              }}
              slidesPerView={1}
            >
              {data &&
                data.length !== 0 &&
                data.map((item, index) => {
                  const { comentario, autor, lugar } = item;

                  return (
                    <SwiperSlide key={index} className={styles.slide}>
                      <Grid item className={styles.comentario}>
                        <Typography
                          className={styles.comentarioText}
                          color="secondary"
                          component="h4"
                          variant="h4"
                        >
                          <Markdown>{comentario}</Markdown>
                        </Typography>
                      </Grid>
                      <Typography className={styles.autor} variant="h2">
                        {autor}
                      </Typography>
                      <Typography className={styles.lugar} variant="h2">
                        {lugar}
                      </Typography>
                    </SwiperSlide>
                  );
                })}
            </SwiperComponent>
          </Grid>
        </Box>
      </Grid>
    </Grid>
  );
};

export default Testimonials;
