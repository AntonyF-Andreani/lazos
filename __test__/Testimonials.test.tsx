import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Testimonials, { TestimonialsProps } from "sections/Testimonials";

const defaultProps: TestimonialsProps = {
  title: "Title test",
  image: { filename: `background-image`, alt: "test" },
  data: [
    {
      _uid: "1",
      comentario: "comentario test",
      autor: "autor test",
      lugar: "test",
    },
    {
      _uid: "2",
      comentario: "comentario test",
      autor: "autor test",
      lugar: "test",
    },
    {
      _uid: "3",
      comentario: "comentario test",
      autor: "autor test",
      lugar: "test",
    },
  ],
};

describe("<Testimonials />", () => {
  it("should render without crash", () => {
    const { asFragment } = render(<Testimonials {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have title", () => {
    const { getByText } = render(<Testimonials {...defaultProps} />);
    const element = getByText(defaultProps.title as string);

    expect(element).toBeInTheDocument();
  });

  it("should have the comentaries", () => {
    const { getAllByText } = render(<Testimonials {...defaultProps} />);

    const elements = getAllByText("comentario test");

    elements.map((el) => {
      return expect(el).toBeInTheDocument;
    });
  });
  it("should have the correct image", () => {
    const { getAllByRole } = render(<Testimonials {...defaultProps} />);

    const elements = getAllByRole("img");

    // loop property Swiper duplicate each logo img
    elements.map((el) => {
      return expect(el.getAttribute("src")).toBe(defaultProps.image.filename);
    });
  });
});
