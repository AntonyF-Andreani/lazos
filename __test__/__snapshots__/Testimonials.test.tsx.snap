// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Testimonials /> should render without crash 1`] = `
<DocumentFragment>
  <section
    class="MuiGrid-root root css-vj1n65-MuiGrid-root"
  >
    <div
      class="mainBanner MuiBox-root css-0"
    >
      <span
        class=" lazy-load-image-background blur"
        style="display: inline-block;"
      >
        <img
          alt="test"
          class="image"
          src="background-image"
        />
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="comentarioContainer MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 title css-1sra7t5-MuiTypography-root"
        >
          Title test
        </h2>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="swiper-container swiper-container-initialized swiper-container-horizontal swiperContainer"
          >
            <div
              class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets"
            />
            <div
              class="swiper-wrapper"
              style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
            >
              <div
                class="swiper-slide swiper-slide-duplicate slide"
                data-swiper-slide-index="2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item comentario css-13i4rnv-MuiGrid-root"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 comentarioText css-i0231a-MuiTypography-root"
                  >
                    <p>
                      comentario test
                    </p>
                  </h4>
                </div>
                <h2
                  class="MuiTypography-root MuiTypography-h2 autor css-1sra7t5-MuiTypography-root"
                >
                  autor test
                </h2>
                <h2
                  class="MuiTypography-root MuiTypography-h2 lugar css-1sra7t5-MuiTypography-root"
                >
                  test
                </h2>
              </div>
              <div
                class="swiper-slide slide"
                data-swiper-slide-index="0"
              >
                <div
                  class="MuiGrid-root MuiGrid-item comentario css-13i4rnv-MuiGrid-root"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 comentarioText css-i0231a-MuiTypography-root"
                  >
                    <p>
                      comentario test
                    </p>
                  </h4>
                </div>
                <h2
                  class="MuiTypography-root MuiTypography-h2 autor css-1sra7t5-MuiTypography-root"
                >
                  autor test
                </h2>
                <h2
                  class="MuiTypography-root MuiTypography-h2 lugar css-1sra7t5-MuiTypography-root"
                >
                  test
                </h2>
              </div>
              <div
                class="swiper-slide slide"
                data-swiper-slide-index="1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item comentario css-13i4rnv-MuiGrid-root"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 comentarioText css-i0231a-MuiTypography-root"
                  >
                    <p>
                      comentario test
                    </p>
                  </h4>
                </div>
                <h2
                  class="MuiTypography-root MuiTypography-h2 autor css-1sra7t5-MuiTypography-root"
                >
                  autor test
                </h2>
                <h2
                  class="MuiTypography-root MuiTypography-h2 lugar css-1sra7t5-MuiTypography-root"
                >
                  test
                </h2>
              </div>
              <div
                class="swiper-slide slide"
                data-swiper-slide-index="2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item comentario css-13i4rnv-MuiGrid-root"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 comentarioText css-i0231a-MuiTypography-root"
                  >
                    <p>
                      comentario test
                    </p>
                  </h4>
                </div>
                <h2
                  class="MuiTypography-root MuiTypography-h2 autor css-1sra7t5-MuiTypography-root"
                >
                  autor test
                </h2>
                <h2
                  class="MuiTypography-root MuiTypography-h2 lugar css-1sra7t5-MuiTypography-root"
                >
                  test
                </h2>
              </div>
              <div
                class="swiper-slide swiper-slide-duplicate slide"
                data-swiper-slide-index="0"
              >
                <div
                  class="MuiGrid-root MuiGrid-item comentario css-13i4rnv-MuiGrid-root"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 comentarioText css-i0231a-MuiTypography-root"
                  >
                    <p>
                      comentario test
                    </p>
                  </h4>
                </div>
                <h2
                  class="MuiTypography-root MuiTypography-h2 autor css-1sra7t5-MuiTypography-root"
                >
                  autor test
                </h2>
                <h2
                  class="MuiTypography-root MuiTypography-h2 lugar css-1sra7t5-MuiTypography-root"
                >
                  test
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
